homeassistant/components/microbot_push/__init__.py

```diff
@@ -0,0 +1,176 @@
+"""
+Custom integration to integrate MicroBot with Home Assistant.
+
+For more details about this integration, please refer to
+https://github.com/spycle/microbot_push
+"""
+from __future__ import annotations
+
+import asyncio
+import logging
+import re
+from typing import TYPE_CHECKING, Any
+
+from microbot import (  # pylint: disable=import-error
+    MicroBotApiClient,
+    parse_advertisement_data,
+)
+
+from homeassistant.components import bluetooth
+from homeassistant.components.bluetooth.passive_update_coordinator import (
+    PassiveBluetoothDataUpdateCoordinator,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, ServiceCall, callback
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import (
+    CONF_BDADDR,
+    CONF_RETRY_COUNT,
+    DEFAULT_RETRY_COUNT,
+    DOMAIN,
+    PLATFORMS,
+    STARTUP_MESSAGE,
+)
+
+if TYPE_CHECKING:
+    from bleak.backends.device import BLEDevice
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up this integration using UI."""
+    hass.data.setdefault(DOMAIN, {})
+    _LOGGER.debug(STARTUP_MESSAGE)
+
+    if not entry.options:
+        hass.config_entries.async_update_entry(
+            entry,
+            options={CONF_RETRY_COUNT: DEFAULT_RETRY_COUNT},
+        )
+
+    bdaddr: Any | None = entry.data.get(CONF_BDADDR)
+    assert bdaddr is not None
+    ble_device: BLEDevice | None = bluetooth.async_ble_device_from_address(
+        hass, bdaddr.upper()
+    )
+    if not ble_device:
+        raise ConfigEntryNotReady(f"Could not find MicroBot with address {bdaddr}")
+    conf_dir = hass.config.path()
+    conf = (
+        conf_dir
+        + "/.storage/microbot-"
+        + re.sub("[^a-f0-9]", "", bdaddr.lower())
+        + ".conf"
+    )
+    client = MicroBotApiClient(
+        device=ble_device,
+        config=conf,
+        retry_count=entry.options[CONF_RETRY_COUNT],
+    )
+    coordinator = MicroBotDataUpdateCoordinator(
+        hass, client=client, ble_device=ble_device
+    )
+
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+
+    for platform in PLATFORMS:
+        coordinator.platforms.append(platform)
+        hass.async_add_job(
+            hass.config_entries.async_forward_entry_setup(entry, platform)
+        )
```

Comment:
Please await platform setups so there is no race where the integration gets reloaded while setup is in progress

Reply:
You can use
```
    await hass.config_entries.async_forward_entry_setups(
        entry, --PLATFORMS HERE--
    )
```